packages/aggregator/src/analytics/aggregateAnalytics/aggregations/getFinalValuePerPeriod.js

```diff
@@ -106,25 +106,26 @@ class FinalValueAggregator {
     this.cache = cache;
   }
 
-  getContinuousValues(analytics, aggregationPeriod) {
-    const periods = getContinuousPeriodsForAnalytics(analytics, aggregationPeriod, true);
+  getFilledValues(analytics, aggregationPeriod, fillEmptyPeriodsWith) {
+    const checkIfFillingContinuousValues = fillEmptyPeriodsWith === 'previous';
+    const periods = getContinuousPeriodsForAnalytics(
+      analytics,
+      aggregationPeriod,
+      checkIfFillingContinuousValues,
+    );
 
     const values = [];
     this.cache.iterateOrganisationUnitCache(organisationUnitCache => {
       let mostRecentValue;
 
       periods.forEach(period => {
-        const valueForPeriod = organisationUnitCache[period];
-
-        if (valueForPeriod !== undefined) {
-          mostRecentValue = valueForPeriod;
-        }
-        if (mostRecentValue !== undefined) {
-          values.push({
-            ...mostRecentValue,
-            period,
-          });
-        }
+        const valueForPeriodFromCache = organisationUnitCache[period];
+        values.push({
+          ...(checkIfFillingContinuousValues && mostRecentValue),
```

Comment:
Here we couldn't specify the default value because we don't have dataElement and organisationUnit like others do. But a period with nothing, the frontend will know there should ba a gap. Values example: 

> [
    {
  dataElement: 'POP21AND23',
  organisationUnit: 'TO_Vainihc',
  period: '201911',
  value: 398
},
{
  period: '201912'
} // show gap
]

Reply:
We wanna return:
 ```
 {
dataElement: 'POP21AND23',
organisationUnit: 'TO_Vainihc',
period: '201911',
value: null
}
```